Adapter/Algolia.php

```diff
@@ -90,140 +96,200 @@ public function __construct(
     {
-        return
-            $this->config->getApplicationID($storeId)
-            && $this->config->getAPIKey($storeId)
-            && $this->config->isEnabledFrontEnd($storeId)
-            && $this->config->makeSeoRequest($storeId);
+        $contextParams = [];
+        $targetedIndex = null;
+        if ($this->isReplaceCategory($storeId)) {
+            $contextParams = $this->getContextParams($storeId);
+
+            if (!is_null($this->request->getParam('sortBy'))) {
+                $targetedIndex = $this->request->getParam('sortBy');
+            }
+        }
+
+        return $this->algoliaHelper->getSearchResult($algoliaQuery, $storeId, $contextParams, $targetedIndex);
     }
 
     /**
-     * Get aggregations
+     * Get the contextual params from the url
      *
-     * @param  RequestInterface $request
-     * @param  Table            $table
-     * @param  array            $documents
+     * @param int $storeId
      *
      * @return array
      */
-    private function getAggregations($request, $table, $documents)
+    private function getContextParams($storeId)
     {
-        if (version_compare($this->config->getMagentoVersion(), '2.1.0', '<') === true) {
-            return $this->aggregationBuilder->build($request, $table);
-        }
+        $contextParams = [];
+        $category = $this->registry->registry('current_category');
+        $page = !is_null($this->request->getParam('page')) ?
+            (int) $this->request->getParam('page') - 1 :
+            0;
+
+        if ($category) {
+            $contextParams = [
+                'facetFilters' => [
+                    'categoryIds:' . $category->getEntityId(),
+                ],
+                'page' => $page,
+            ];
+
+            $facetFilters = [];
+
+            foreach ($this->config->getFacets($storeId) as $facet) {
+                if (!is_null($this->request->getParam($facet['attribute']))) {
+                    if ($facet['type'] === 'conjunctive') {
+                        $facetValues = explode('~', $this->request->getParam($facet['attribute']));
+                        foreach ($facetValues as $key => $facetValue) {
+                            $facetFilters[] = $facet['attribute'] . ':' . $facetValue;
+                        }
+                    }
+
+                    if ($facet['type'] === 'disjunctive') {
+                        $facetValues = explode('~', $this->request->getParam($facet['attribute']));
+                        if (count($facetValues) > 1) {
+                            foreach ($facetValues as $key => $facetValue) {
+                                $facetValues[$key] = $facet['attribute'] . ':' . $facetValue;
+                            }
+                            $facetFilters[] = $facetValues;
+                        }
+                        if (count($facetValues) == 1) {
+                            $facetFilters[] = $facet['attribute'] . ':' . $facetValues[0];
```

Comment:
It's not that easy with `disjunctive` faceting in PHP. Disjunctive facets need to be sent in different query paramter: https://github.com/algolia/algoliasearch-client-php/blob/master/src/AlgoliaSearch/Index.php#L614